assets/javascripts/discourse-assign/initializers/extend-for-assigns.js.es6

```diff
@@ -312,13 +314,67 @@ function initialize(api) {
   api.addKeyboardShortcut("g a", "", { path: "/my/activity/assigned" });
 }
 
+export const REGEXP_USERNAME_PREFIX = /^(assigned:)/gi;
+
 export default {
   name: "extend-for-assign",
   initialize(container) {
     const siteSettings = container.lookup("site-settings:main");
     if (!siteSettings.assign_enabled) {
       return;
     }
+    const currentUser = container.lookup("current-user:main");
+    if (currentUser.can_assign) {
+      SearchAdvancedOptions.reopen({
+        _init() {
```

Comment:
Instead of overwriting the `SearchAdvancedOptions._init()` method, we should add a new "plugin API" in core to allow plugin to add new "in options".